odbc/src/api/get_diag_field_tests.rs

```diff
@@ -0,0 +1,348 @@
+                    0,
+                    DiagType::MessageText,
+                    (&mut [0u16; 6]) as *mut _ as *mut c_void,
+                    57,
+                    &mut 0
+                )
+            );
+            // Record number > number of diagnostic records
+            assert_eq!(
+                SqlReturn::NO_DATA,
+                SQLGetDiagFieldW(
+                    HandleType::Env,
+                    env_handle as *mut _,
+                    3,
+                    DiagType::MessageText,
+                    (&mut [0u16; 6]) as *mut _ as *mut c_void,
+                    57,
+                    &mut 0
+                )
+            );
+            // Header fields that require a statement handle
+            assert_eq!(
+                SqlReturn::ERROR,
+                SQLGetDiagFieldW(
+                    HandleType::Env,
+                    env_handle as *mut _,
+                    1,
+                    DiagType::RowCount,
+                    (&mut 0) as *mut _ as *mut c_void,
+                    10,
+                    &mut 0
+                )
+            );
+            assert_eq!(
+                SqlReturn::ERROR,
+                SQLGetDiagFieldW(
+                    HandleType::Env,
+                    env_handle as *mut _,
+                    1,
+                    DiagType::RowNumber,
+                    (&mut 0) as *mut _ as *mut c_void,
+                    10,
+                    &mut 0
+                )
+            );
+        }
+    }
+
+    /*
+       TODO: This test should be removed when we implement the rest of the diag field types
+    */
+    #[test]
+    fn test_invalid_diag_identifier() {
+        let env_handle: *mut _ = &mut MongoHandle::Env(Env::with_state(EnvState::Allocated));
+
+        unsafe {
+            (*env_handle).add_diag_info(ODBCError::Unimplemented("SQLDrivers"));
+
+            // check first we have exactly one error in the handle
+            let num_errors_buffer = &mut 0u64 as *mut _ as *mut c_void;
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetDiagFieldW(
+                    HandleType::Env,
+                    env_handle as *mut _,
+                    0,
+                    DiagType::Number,
+                    num_errors_buffer,
+                    0,
+                    &mut 0
+                )
+            );
+            assert_eq!(1, *(num_errors_buffer as *const u64));
+
+            // make a call for an unimplemented diagnostic type
+            assert_eq!(
+                SqlReturn::ERROR,
+                SQLGetDiagFieldW(
+                    HandleType::Env,
+                    env_handle as *mut _,
+                    1,
+                    DiagType::DynamicFunctionCode,
+                    &mut 0 as *mut _ as *mut c_void,
+                    1,
+                    &mut 0
+                )
+            );
+            // check the correct diagnostic was added for this error
```

Comment:
The code below will not be needed anymore if/when you remove the code which adds a error record on a failed SQLGetDiagRec call.
If you think that it makes sense to keep adding it, we can also discuss that during standup. I'm open to having more views on it.